design/repo_maintenance_job_config.md

```diff
@@ -0,0 +1,144 @@
+
+## Abstract
+Add this design to make the repository maintenance job can read configuration from a dedicate ConfigMap and make the Job's `PodSpec.Affinity` configurable. 
+
+## Background
+Repository maintenance is split from the Velero server to a k8s Job in v1.14 by [repository maintenance job](Implemented/repository-maintenance.md).
+The repository maintenance Job configuration was read from the Velero server CLI parameter.
+The Job inherits the most of the Job's PodSpec configuration from Velero server's Deployment's PodSpec.
+That inheritance also includes the Affinity setting.
+
+[node-agent affinity configuration](Implemented/node-agent-affinity.md) introduces affinity configuration for the node-agent DaemonSet pods.
+This design reuses the data structure introduced by that design to make the repository maintenance job can choose which node running on.
+
+## Goals
+- Let user can choose repository maintenance Job running on which nodes.
+- Unify the repository maintenance Job configuration at one place.
+
+## Non Goals
+- There was an [issue](https://github.com/vmware-tanzu/velero/issues/7911) to require the whole Job's PodSpec should be configurable. That's not in the scope of this design.
+- The introduced configuration is universal for all repositories. Configuration dedicated for each repository is not in this design's scope.
+
+## Alternatives Considered
+An other option is making each BackupRepository having its own repository maintenance job configuration.
+Each BackupRepository has its own storage capacity, so they may require different resource to maintain. 
+This can be done by adding a field to the BackupRepository CRD to specify the referenced ConfigMap.
+After discussing, we think there is no need to make the new configuration for each BackupRepository.
+The configuration should be specific for the repository maintenance job.
+
+## Compatibility
+v1.14 uses the `velero server` CLI's parameter to pass the repository maintenance job configuration.
+If the ConfigMap labelled as `velero.io/config-map-type=repo-maintenance-job-config` doesn't exist, those parameters are still respected.
+If the ConfigMap labelled as `velero.io/config-map-type=repo-maintenance-job-config` exists, the ConfigMap provided parameters override the CLI parameters.
+
+## Design
+This design introduces a new ConfigMap labelled as `velero.io/config-map-type=repo-maintenance-job-config` as the source of the repository maintenance job configuration. 
+If the `repo-maintenance-job-config` doesn't exist, the `velero server` parameters related to the repository maintenance job are used.
+
+**Notice**
+* Velero doesn't own this ConfigMap. If the user wants to customize the repository maintenance job, the user needs to create this ConfigMap.
+* Velero reads this ConfigMap content at start, so the ConfigMap change after the Velero server start cannot be reflected. To make the change take affect, please restart the Velero Deployment pod.
```

Comment:
I don't think it is necessary. The configMap could be visited by the backupRepo controller at the time of starting a repo maintenance job, so users don't need to restart the Velero server in order to make it effective.